swim/memberlist.go

```diff
@@ -324,3 +324,17 @@ func (m *memberlist) String() string {
 func (m *memberlist) Iter() *memberlistIter {
 	return newMemberlistIter(m)
 }
+
+func (m *memberlist) GetActiveMemberAddresses() []string {
+	var active []string
+
+	m.members.RLock()
```

Comment:
I chose this over copying the list because a copy operation allocates new memory which we don't need. I don't know what the time difference between a copy and the iteration is. But now I run over the list once and only append the members I need to the new list. I'm also fine with copying the list and iterate over a copy.

First I used the accessors to the memberlist, but every operation tries to acquire the lock for its self. Here I just lock the list once, iterate over it and release the lock.